chore(docs): Update tutorial to Head API (#36378)
gatsbyjs/gatsby: docs/docs/tutorial/part-2/index.mdx

By the end of this part of the Tutorial, you will be able to:

* Create **page components** to add new pages to your site.

> * Add title to your site using the **Gatsby Head API**

Reviewer: ```suggestion
* Add a title to your site using the **Gatsby Head API**
```

const IndexPage = () => {
  return (
    <main>
      <h1>Welcome to my Gatsby site!</h1>
      <p>I'm making this by following the Gatsby Tutorial.</p>
    </main>
  )
}

> // Page title that shows up on the browser tab

Reviewer: ```suggestion
// Page title that shows up on the browser tab and in the <head> element
```

export default AboutPage
```

2. Add a page title to your page. Gatsby lets you define a `<title>` and other [document metadata](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) with the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/). You have to export a component called `Head` from your page to apply the metadata.

> Adding such metadata helps search engines like Google to better understand your site. For this tutorial we'll only be adding a title to this page but you can also later add other metadata.

Reviewer: ```suggestion
Adding such metadata helps search engines like Google to better understand your site. For this tutorial you'll only be adding a title to this page but you can also later add other metadata.
```

```js:title=src/pages/about.js
import * as React from 'react'

const AboutPage = () => {
  return (
    <main>
      <h1>About Me</h1>
      <p>Hi there! I'm the proud creator of this site, which I built with Gatsby.</p>
    </main>
  )
}

> // Yay! We defined a page title. This shows up on the browser tab

Reviewer: ```suggestion
// Yay! You defined a page title. This shows up on the browser tab
```

export const Head = () => <title>About Me<title/>

export default AboutPage

> ```

Reviewer: Add a "Key Gatsby Concept" below this second point here about the Head API

Author: 👍

export default AboutPage
```

> 2. Add a page title: Gatsby lets you define a title and other [document head](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) related tags with the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/).
> When you export a `Head` component from your page file, It automatically picks all the head tags in your JSX and adds them to the [document head](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head).
> This is important because it helps search engines like Google to better understand your site. For this tutorial we'll only be adding a title to this page but you can also add `<meta>`, `<links>` and other head related tags

Reviewer: ```suggestion
2. Add a page title to your page. Gatsby lets you define a `<title>` and other [document metadata](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) with the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/). You have to export a component called `Head` from your page to apply the metadata.
Adding such metadata helps search engines like Google to better understand your site. For this tutorial we'll only be adding a title to this page but you can also later add other metadata.
```

Author: Shorter, I like it 👍

```js:title=src/pages/about.js
import * as React from 'react'

const AboutPage = () => {
  return (
    <main>
      <h1>About Me</h1>
      <p>Hi there! I'm the proud creator of this site, which I built with Gatsby.</p>
    </main>
  )
}

// Yay! We defined a page title. This shows up on the browser tab
export const Head = () => <title>About Me<title/>

export default AboutPage
```

> 3. In a web browser, visit `localhost:8000/about`. When your development server finishes rebuilding your site, the About page should look something like this:

Reviewer: The "Key Gatsby Concept" below should also contain a mention of the Gatsby Head API

Author: Each of the key concepts down there (from here https://www.gatsbyjs.com/docs/tutorial/part-2/#use-the-link-component) seem to be talking about specific things like Link, Layout, and Stying, not sure where to slot that.

If we decide to create a new heading for the Head API then we'd have to elaborate and also stay consistent by adding a collapsible like  👇

<img width="785" alt="Screenshot 2022-08-16 at 07 13 51" src="https://user-images.githubusercontent.com/17142206/184810581-83857452-1cb4-41d2-a415-c173fd27f09a.png">

export default AboutPage
```

2. Add a page title to your page. Gatsby lets you define a `<title>` and other [document metadata](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/head) with the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/). You have to export a component called `Head` from your page to apply the metadata.
Adding such metadata helps search engines like Google to better understand your site. For this tutorial you'll only be adding a title to this page but you can also later add other metadata.

```js:title=src/pages/about.js
import * as React from 'react'

const AboutPage = () => {
  return (
    <main>
      <h1>About Me</h1>
      <p>Hi there! I'm the proud creator of this site, which I built with Gatsby.</p>
    </main>
  )
}

// Yay! You defined a page title. This shows up on the browser tab
export const Head = () => <title>About Me<title/>

export default AboutPage
```
<Announcement style={{marginBottom: "1.5rem"}}>

**Key Gatsby Concept** 💡

> You just added a title to the About page using the [Gastby Head API](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/). You'll notice that the title is now displayed on the browser tab.
> With this API, you can also add other metadata to your site. For example, you can add a description to your site that shows up in search engines like Google.

Reviewer: ````suggestion
You can use the [Gatsby Head API](/docs/reference/built-in-components/gatsby-head/) by exporting a named function called `Head` in your pages and templates that are used to create pages. 

Be sure to capitalize `Head` and please note that exporting this named function inside a component won't add the metadata to the `<head>`. The above works because you're exporting `Head` in a page inside `src/pages`.

You can add any valid `<head>` tags inside the function and they'll be added to the page, for example:

```js
export const Head = () => (
  <>
    <title>About Me<title/>
    <meta name="description" content="Your description" />
  </>
)
```

After going through this tutorial, be sure to check out [Adding an SEO Component](/docs/how-to/adding-common-features/adding-seo-component/).
````

  )
}

> export const Head = () => <title>About Page</title>

Reviewer: ```suggestion
export const Head = () => <title>About Me</title>
```

* What's the difference between a page component and a building-block component?
* How do you add a new page to your Gatsby site?

> * How do you add title to page?

Reviewer: ```suggestion
* How do you add a title to a page?
```
